crates/python/src/api.rs

```diff
@@ -102,6 +102,16 @@ py_wrap_error!(
     PyRuntimeError
 );
 
+/// Get the keyword `key` value from `kwds` if it is of type `Option<T>`, else `None`.
+/// Note that values present at `key` that cannot be extracted into `Option<T>` are discarded.
+fn get_kwd<'a, T: FromPyObject<'a>>(kwds: Option<&'a PyDict>, key: &str) -> Option<T> {
+    if let Some(kwds) = kwds {
+        kwds.get_item(key).and_then(|value| value.extract().ok())
+    } else {
+        None
+    }
```

Comment:
```suggestion
    kwds
        .and_then(|kwds| kwds.get_item(key))
        .and_then(|value| value.extract().ok())
```

Alternatively, change this to returning a `PyResult<Option<T>>` and

```suggestion
    kwds
        .and_then(|kwds| kwds.get_item(key))
        .map(|value| value.extract())
        .transpose()
```

Reply:
FWIW, I think returning a `Result` is preferred, because it should probably be an error if the wrong type is provided, rather than ignoring that anything was provided in the first place.